src/Link.ts

```diff
@@ -10,25 +10,41 @@ export interface LinkProperties extends WidgetProperties, VirtualDomProperties {
 	isOutlet?: boolean;
 	params?: any;
 	routerKey?: string;
+	onClick?: (event: MouseEvent) => void;
 	to: string;
 }
 
 export class Link extends WidgetBase<LinkProperties> {
 
+	private _onClick(event: MouseEvent): void {
+		this.properties.onClick && this.properties.onClick(event);
+	}
+
 	@beforeRender()
 	protected withRouter(renderFunc: () => DNode, properties: any, children: any): () => DNode {
-		const { to, isOutlet = true, params = {}, routerKey = globalRouterKey, ...props } = properties;
+		const { to, isOutlet = true, params = {}, routerKey = globalRouterKey, onClick, ...props } = properties;
 		if (this.registries.get(routerKey)) {
 			return () => w<RouterInjector>(routerKey, {
 				scope: this,
 				render: renderFunc,
 				properties,
 				getProperties: (router: Router<any>, properties: LinkProperties) => {
-					if (!isOutlet) {
-						return properties;
-					}
+					const handleOnClick = (event: MouseEvent) => {
+						const { to } = this.properties;
+
+						if (onClick) {
```

Comment:
We should probably standardise, but I'm pretty guilty at flip flopping between the two.